simulator/modules/branch/branch.cpp

```diff
@@ -47,8 +47,15 @@ void Branch<FuncInstr>::clock( Cycle cycle)
     /* acquiring real information for BPU */
     wp_bp_update->write( instr.get_bp_upd(), cycle);
      
+    bool is_misprediction = false;
+
+    if ( ( instr.get_bp_data() ).is_taken != instr.is_taken())
```

Comment:
Please remove excessive brackets around the `instr.get_bp_data()` here and below